SimpleITK/utilities/resize_and_scale.py

```diff
@@ -0,0 +1,54 @@
+import SimpleITK as sitk
+
+
+def resize_and_scale(image, new_size, outside_pixel_value=0):
+    """
+    Resize the given image to the given size, with isotropic pixel spacing
+    and scale the intensities to [0,255].
+
+    Resizing retains the original aspect ratio, with the original image centered
+    in the new image. Padding is added outside the original image extent using the
+    provided value.
+
+    :param image: A SimpleITK image.
+    :param new_size: List of ints specifying the new image size.
+    :param outside_pixel_value: Value in [0,255] used for padding.
+    :param preserve_aspect_ratio:
+    :return: a 2D SimpleITK image with desired size and a pixel type of sitkUInt8
+    """
+    # Rescale intensities if scalar image with pixel type that isn't sitkUInt8.
+    # We rescale first, so that the zero padding makes sense for all original image
+    # ranges. If we resized first, a value of zero in a high dynamic range image may
+    # be somewhere in the middle of the intensity range and the outer border has a
+    # constant but arbitrary value.
+    if (
+        image.GetNumberOfComponentsPerPixel() == 1
+        and image.GetPixelID() != sitk.sitkUInt8
+    ):
+        final_image = sitk.Cast(sitk.RescaleIntensity(image), sitk.sitkUInt8)
```

Comment:
The casting is not necessary, as the output pixel type for resample can be set. 

Also the rescaling will not work when converting  int8 to uint8.

It may be preferred to do ShiftScale which allows setting the output pixel type while doing the intensity shift.